src/document/printer.js

```diff
@@ -292,283 +317,292 @@ function printDocToString(doc, options) {
 
   while (cmds.length > 0) {
     const [ind, mode, doc] = cmds.pop();
+    const type = getDocType(doc);
 
-    if (typeof doc === "string") {
-      const formatted = newLine !== "\n" ? doc.replace(/\n/g, newLine) : doc;
-      out.push(formatted);
-      pos += getStringWidth(formatted);
-    } else if (isConcat(doc)) {
-      const parts = getDocParts(doc);
-      for (let i = parts.length - 1; i >= 0; i--) {
-        cmds.push([ind, mode, parts[i]]);
+    switch (type) {
```

Comment:
Let's do `switch(getDocType(doc))` instead?

Reply:
I was going to use type in default branch, but I don't need that anymore.